### PR TITLE
You can sacrifice souls to toolboxes

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Back in No Feature November, Swanni on the /tg/ discord had a funny idea

[](https://i.imgur.com/QAPIxW0.png)

So, now that it's December, this PR makes it so you can use a soulstone with a shade in it on a mechanical toolbox to sacrifice the shade to give the toolbox back its SOUL (it goes back to the old sprite)

[](https://i.imgur.com/2a3SR4A.png)

Also adds an achievement for being the poor shade who gets sacrificed to the toolbox, "SOUL'd Out"

## Why It's Good For The Game
Destroying someone's eternal soul in order to make a toolbox look different is one of the most powerful forms of BM I can imagine

## Changelog
:cl: Ryll/Shaps
add: You can now sacrifice a shade in a soulstone to reclaim a mechanical toolbox's SOUL
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/toolbox.dm [hunk 1]
```diff
@@ -84,7 +86,7 @@
 	name = "rusty blue toolbox"
 	icon_state = "toolbox_blue_old"
 	has_latches = FALSE
-	material_flags = NONE
```
Comment (Fikou) [inline code/game/objects/items/storage/toolbox.dm:87]: ?

File: code/modules/antagonists/wizard/equipment/soulstone.dm [hunk 2]
```diff
@@ -114,12 +114,39 @@
...
+/obj/item/soulstone/pre_attack(atom/A, mob/living/user, params)
+	var/mob/living/simple_animal/shade/occupant = (locate() in src)
+	var/obj/item/storage/toolbox/mechanical/target_toolbox = A
+	if(!occupant || !istype(target_toolbox) || target_toolbox.has_soul)
+		return ..()
+
+	if(!iscultist(user) && !iswizard(user) && !usability)
+		user.Unconscious(10 SECONDS)
+		to_chat(user, "<span class='userdanger'>Your body is wracked with debilitating pain!</span>")
+		return
```
Comment (ATH1909) [inline code/modules/antagonists/wizard/equipment/soulstone.dm:133]: don't forget the hot potato check for cultists using purified soulstones

**Comment by ATH1909** [comment]
What happens if you use this on a yellow or red toolbox?

**Comment by ATH1909** [comment]
Would it be possible to move the shade into the toolbox (like how they are for things like possessed blades and soulstones, IIRC), allowing the shade who was sacrificed to the toolbox to still speak (as the toolbox's soul)? Would that also open up the possibility of destroying the toolbox to free the shade trapped within it?

File: code/game/objects/items/storage/toolbox.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/game/objects/items/storage/toolbox.dm:87]: It's the same as the parent's var so it's redundant to have here

**Comment by Ryll-Ryll** [comment]
It only works on the blue mechanical toolboxes because (aside from red emergency toolboxes I guess) it's the only one that has the old sprite in the files. As for trapping souls in the toolbox, that'd require additional structure and support on the mechanical toolboxes, and sacrificing the soul completely destroying the shade seemed a bit funnier in my mind (though both a soul stuck in a toolbox and a soul destroyed for the toolbox are funny)

**Comment by Ghommie** [comment]
You can always scroll through commit history to get the other old toolbox sprites from 4-5 years ago.

**Comment by jacklecroy** [comment]
> 
> 
> It only works on the blue mechanical toolboxes because (aside from red emergency toolboxes I guess) it's the only one that has the old sprite in the files. As for trapping souls in the toolbox, that'd require additional structure and support on the mechanical toolboxes, and sacrificing the soul completely destroying the shade seemed a bit funnier in my mind (though both a soul stuck in a toolbox and a soul destroyed for the toolbox are funny)

Can't you just move the shade into the toolbox?

**Comment by Ryll-Ryll** [comment]
> > It only works on the blue mechanical toolboxes because (aside from red emergency toolboxes I guess) it's the only one that has the old sprite in the files. As for trapping souls in the toolbox, that'd require additional structure and support on the mechanical toolboxes, and sacrificing the soul completely destroying the shade seemed a bit funnier in my mind (though both a soul stuck in a toolbox and a soul destroyed for the toolbox are funny)
> 
> Can't you just move the shade into the toolbox?

I checked out the possessed scythe the chap can have and was surprised that this is in fact how it handles being possessed, but the issue is that toolboxes are storage containers, which very likely wouldn't play well with the storage component. As such, I'll pass for now

**Comment by jacklecroy** [comment]
> I checked out the possessed scythe the chap can have and was surprised that this is in fact how it handles being possessed, but the issue is that toolboxes are storage containers, which very likely wouldn't play well with the storage component. As such, I'll pass for now

It shouldn't affect the storage component at all.

**Comment by Ryll-Ryll** [comment]
@jacklecroy Sorry pal, just tested it and the toolbox had one less slot available when the shade was inside of it, and that slot opened back up once I deleted the shade. If you'd like to expand on this PR in the future you're free to, but I'm set with how this is

**Review by Timberpoes — APPROVED** [review]

**Pull request merged**